Sources_v3/StreamChatUI/ChatChannelList/ChatChannelListCollectionViewCell.swift

```diff
@@ -5,4 +5,62 @@
 import Foundation
 import UIKit
 
-open class ChatChannelListCollectionViewCell: UICollectionViewCell {}
+open class ChatChannelListCollectionViewCell<ExtraData: UIExtraDataTypes>: UICollectionViewCell {
+    // MARK: - Properties
+    
+    public weak var collectionView: UICollectionView?
+    
+    public var channelView: ChatChannelView<ExtraData>? {
+        didSet {
+            oldValue?.removeFromSuperview()
+            if let new = channelView {
+                contentView.embed(new)
+            }
+        }
+    }
+    
+    // MARK: - Init
+    
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+
+    private func commonInit() {
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            contentView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            contentView.topAnchor.constraint(equalTo: topAnchor),
+            contentView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+    }
+    
+    // MARK: - Layout
+    
+    override open func preferredLayoutAttributesFitting(
+        _ layoutAttributes: UICollectionViewLayoutAttributes
+    ) -> UICollectionViewLayoutAttributes {
+        let attributes = super.preferredLayoutAttributesFitting(layoutAttributes)
+        
+        if let collectionViewSize = collectionView?.bounds.size {
+            let targetSize = CGSize(
+                width: collectionViewSize.width,
+                height: UIView.layoutFittingCompressedSize.height
+            )
+            
+            attributes.frame.size = contentView.systemLayoutSizeFitting(
+                targetSize,
+                withHorizontalFittingPriority: .required,
+                verticalFittingPriority: .defaultLow
+            )
+        }
+        
+        return attributes
+    }
```

Comment:
As we discussed the cells should be **self-sizing** in order to grow depending on the content and/or accessibility changes. To have self-sizing cells we need all power of **auto-layout**. 

At the same time we want to have a collection view layout being overridable by end users. Custom collection view layouts usually depend on **manual cell attributes calculation** which doesn't work for us since we need auto-layout.

One of the ways to meet all the requirements is to have a cell that calculates its attributes by itself. We have nice separation of concerns then:
1. `ChannelView` is responsible for the content
2. `ChannelViewCell` is responsible for how the `ChannelView` is laid out
3. `ChanneListLayout` is responsible for relations in-between multiple `ChannelViewCell` (like spacing)

Reply:
updated to set `estimatedItemSize` in layout is being prepared and calculate cell size based on the estimated size